.changeset/witty-candles-whisper.md

```diff
@@ -0,0 +1,49 @@
+---
+'@graphql-hive/gateway-runtime': minor
+'@graphql-hive/gateway': minor
+---
+
+New Retry and Timeout plugins;
+
+- Retry plugin: Retry a request if it fails
+
+It respects the `Retry-After` HTTP header, [See more about this HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After)
+
+```ts
+export const gatewayConfig = defineConfig({
+    upstreamRetry: {
+        // The maximum number of retries to attempt.
+        maxRetries: 3, // required
+        // The delay between retries in milliseconds.
+        retryDelay: 1000, // default
+        /**
+         * A function that determines whether a response should be retried.
+         * If the upstream returns `Retry-After` header, the request will be retried.
+         */
+        shouldRetry: ({ response }) => response?.status >= 500 || response?.status === 429
+    }
+    // or you can configure it by subgraph name
+    upstreamRetry({ subgraphName }) {
+        if (subgraphName === 'my-rate-limited-subgraph') {
+            return {
+                maxRetries: 3,
+            }
+        }
```

Comment:
Perhaps add a default number of retries ? Otherwise it means only the rate limited subgraph is the only one that get retries :-P 
```suggestion
        }
        return { maxRetries: 10 }
```